
syntax = "proto3";

message Request{
    RequestType rtype =1;
    GType gtype =2;
    string status_modification =3;    
    ON_OFF on_off=4;
    
    enum GType{
        TV = 0;
        AR = 1;
        LAMPADA = 2;
    }

    enum RequestType{
        ReadStatus = 0;
        ReadSensor = 1;
        ModStatus  = 2;
        ModOnOf=3;
    }
    
    enum ON_OFF{
        ON =0;
        OFF=1;
    }

}

message Response{
    string object_name =1;
    string object_result =2;
}